bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -848,9 +861,19 @@ public void stop() {
 
 @Override
 public boolean setText(String html, boolean trusted) {
-	char[] data = new char[html.length() + 1];
-	html.getChars(0, html.length(), data, 0);
-	return webView.NavigateToString(data) == COM.S_OK;
+	this.html = html;
+	return setUrl(Browser.BASE_URL, null, null);
+}
+
+private void writeToDefaultPathDOM() {
+	if(Paths.get(URI.create(getUrl()))
+			.equals(Paths.get(URI.create(Browser.BASE_URL))) && html != null) {
```

Comment:
You can test for `equals` on the URIs so you don't need to convert them to Paths (and test `html` for null first so as not create unnecessary URIs):

```java
if(html != null && URI.create(getUrl())
			.equals(URI.create(BASE_URL))) {
```